modules/calib3d/src/chessboard.cpp

```diff
@@ -0,0 +1,3209 @@
+    draw(debug_image,img);
+    std::vector<Ellipse>::iterator iter;
+    for(;iter != ellipses.end();++iter)
+        iter->draw(img);
+    cv::imshow("chessboard",img);
+    cv::waitKey(-1);
+#endif
+}
+
+
+void Chessboard::Board::growLeft()
+{
+    if(isEmpty())
+        CV_Error(Error::StsInternal,"Board is empty");
+    PointIter iter(top_left,TOP_LEFT);
+    std::vector<cv::Point2f> points;
+    cv::Point2f pt;
+    do
+    {
+        PointIter iter2(iter);
+        cv::Point2f *p0 = *iter2;
+        iter2.right();
+        cv::Point2f *p1 = *iter2;
+        iter2.right();
+        cv::Point2f *p2 = *iter2;
+        if(iter2.right())
+            estimatePoint(**iter2,*p2,*p1,*p0,pt);
+        else
+            estimatePoint(*p2,*p1,*p0,pt);
+        points.push_back(pt);
+    }
+    while(iter.bottom());
+    addColumnLeft(points);
+}
+
+bool Chessboard::Board::growLeft(const cv::Mat &map,cv::flann::Index &flann_index)
+{
+#ifdef CV_DETECTORS_CHESSBOARD_DEBUG
+    std::vector<Ellipse> ellipses;
+#endif
+    if(isEmpty())
+        CV_Error(Error::StsInternal,"growLeft: Board is empty");
+    PointIter iter(top_left,TOP_LEFT);
+    std::vector<cv::Point2f> points;
+    int count = 0;
+    Ellipse ellipse;
+    cv::Point2f pt;
+    do
+    {
+        PointIter iter2(iter);
+        cv::Point2f *p0 = *iter2;
+        iter2.right();
+        cv::Point2f *p1 = *iter2;
+        iter2.right();
+        cv::Point2f *p2 = *iter2;
+        cv::Point2f *p3 = NULL;
+        if(iter2.right())
+            p3 = *iter2;
+        if(!estimateSearchArea(*p2,*p1,*p0,CORNERS_SEARCH,ellipse,p3))
+            return false;
+        float result = findMaxPoint(flann_index,map,ellipse,white_angle,black_angle,pt);
+        if(pt == *p0)
+        {
+            ++count;
+            points.push_back(ellipse.getCenter());
+        }
+        else if(result != 0)
+        {
+            points.push_back(pt);
+            if(result < 0)
+                ++count;
+        }
+        else
+        {
+            ++count;
+            if(pt.x != pt.x)    // NaN check
+                points.push_back(ellipse.getCenter());
+            else
+                points.push_back(pt);
+        }
+#ifdef CV_DETECTORS_CHESSBOARD_DEBUG
+        ellipses.push_back(ellipse);
+#endif
+    }
+    while(iter.bottom());
+#ifdef CV_DETECTORS_CHESSBOARD_DEBUG
+    drawEllipses(ellipses);
+#endif
+    if(points.size()-count <= 2)
+        return false;
+    if(count > points.size()*0.5 || !checkRowColumn(points))
+        return false;
+    addColumnLeft(points);
+    return true;
+}
+
+void Chessboard::Board::growTop()
+{
+    if(isEmpty())
+        CV_Error(Error::StsInternal,"Board is empty");
+    PointIter iter(top_left,TOP_LEFT);
+    std::vector<cv::Point2f> points;
+    cv::Point2f pt;
+    do
+    {
+        PointIter iter2(iter);
+        cv::Point2f *p0 = *iter2;
+        iter2.bottom();
+        cv::Point2f *p1 = *iter2;
+        iter2.bottom();
+        cv::Poin
```

Comment:
Not used?

Reply:
There is GitHub bug with code location, again =(

Code is [here](https://github.com/opencv/opencv/pull/12147/files#diff-8df6e9bc9d383d0c31ba9d371e491352R3173) (cv::findChessboardCorners2, line 3173):
```
std::vector<cv::Point2f> out_corners;
```